### PR TITLE
Shipping addresses filter and fixing related to it - https://particl.atlassian.net/browse/PD-157

### PR DESCRIPTION
**Pull request by vikas-cis**

### CODE REVIEW

File: src/app/core/market/api/profile/profile.service.ts [hunk 1]
```diff
@@ -54,7 +56,7 @@ export class ProfileService implements OnDestroy {
   }
 
   get(profileIdOrName: number | string): Observable<any> {
-    return this.market.call('profile', ['get', profileIdOrName]);
+    return this.market.call('profile', ['get', profileIdOrName])
```
Comment (kewde) [inline src/app/core/market/api/profile/profile.service.ts]: map to Profile in get() not in default().

File: src/app/market/buy/checkout-process/checkout-process.component.html [hunk 2]
```diff
@@ -135,8 +135,8 @@
               <mat-select [(ngModel)]="selectedAddress" name="selectedAddress"  [ngModelOptions]="{standalone: true}"
                           (change)="select(selectedAddress)"
                           placeholder="Select saved shipping address profile"
-                          disabled="{{ !profile.ShippingAddresses || profile.ShippingAddresses.length === 0 }}">
-                <mat-option *ngFor="let address of profile.ShippingAddresses"
+                          disabled="{{ !profile?.shippingAddresses || profile?.shippingAddresses.length === 0 }}">
```
Comment (kewde) [inline src/app/market/buy/checkout-process/checkout-process.component.html]: let's use [disabled]="!profile?.shippingAddresses || profile?.shippingAddresses.length === 0"

File: src/app/core/market/api/profile/profile.model.ts [hunk 3]
```diff
@@ -1,6 +1,40 @@
...
+  public shoppingCarts: Array<Cart>;
+  public favorites: Array<Favorite>;
+
+  constructor(private object: any) {
+    this.setShippingAddresses();
+    this.setShoppingCarts();
+    this.setFavoriteItems();
+  }
+
+  get id() {
```
Comment (kewde) [inline src/app/core/market/api/profile/profile.model.ts]: number

File: src/app/core/market/api/profile/profile.model.ts [hunk 4]
```diff
@@ -1,6 +1,40 @@
...
+    return this.object.id;
+  }
+
+  get address() {
```
Comment (kewde) [inline src/app/core/market/api/profile/profile.model.ts]: string

File: src/app/core/market/api/profile/profile.model.ts [hunk 5]
```diff
@@ -1,6 +1,40 @@
...
+    return this.object.address;
+  }
+
+  get name() {
```
Comment (kewde) [inline src/app/core/market/api/profile/profile.model.ts]: string

File: src/app/core/market/api/profile/profile.model.ts [hunk 6]
```diff
@@ -1,6 +1,40 @@
...
+    return this.object.name;
+  }
+
+  setShippingAddresses() {
+    this.shippingAddresses = this.object.ShippingAddresses
+                            .filter((address) => address.title && address.type === 'SHIPPING_OWN')
+  }
+
+  setShoppingCarts() {
+    this.shoppingCarts = this.object.ShoppingCart.filter((shop) => shop.name === 'DEFAULT')
```
Comment (kewde) [inline src/app/core/market/api/profile/profile.model.ts]: Not sure if we want these filters by default.. hmmm, need to think about it

Comment (vikas-cis) [reply src/app/core/market/api/profile/profile.model.ts]: yeah :)

**Comment by vikas-cis** [comment]
@kewde seems the travis build needs restart or clearing the cache

**Pull request merged**